GoldRush.js

```diff
@@ -0,0 +1,116 @@
+class GoldRush extends Matrix {
+  constructor(numRow, numCol) {
+    super(numRow, numCol);
+    this.p1 = {coords :{ x: 0, y: 0 },score : 0};
+    this.p2 = {coords:{x: numCol - 1, y: numRow - 1 },score : 0};
+    this.coins 
+  }
+
+  generateCoins(numCoins){
+      let y, x
+      for(let i = 0; i< numCoins; i++){
+           y = Math.floor(Math.random() * this.matrix.length)
+           x = Math.floor(Math.random() * this.matrix[0].length)
+           if(this.get(y,x) == "."){
+               this.alter(y,x,"C")
+           }
+      }
+      this.coins = numCoins
+  }
+  generateWalls(numWalls){
+    let y, x
+    for(let i = 0; i< numWalls; i++){
+         y = Math.floor(Math.random() * this.matrix.length)
+         x = Math.floor(Math.random() * this.matrix[0].length)
+         if(this.get(y,x) == "." /*&&((this.get(y+1,x) || this.get(y-1,x) || this.get(y,x+1) ||this.
+         get(y,x-1)) == "."*/){
+             this.alter(y,x,"wall")
+         }else{
+           i--
+         }
+    }
+}
+
+  movePlayer(player, direction) {
+    
+    switch (direction) {
+      case "up":
+          if(this.validateMove(player,direction)){
+              this.moveUp(player);
+              break;
+          }
+      case "down":
+        if(this.validateMove(player,direction)){
+            this.moveDown(player);
+            break;
+        }
+      case "left":
+        if(this.validateMove(player,direction)){
+            this.moveLeft(player);
+            break;
+        }
+      case "right":
+        if(this.validateMove(player,direction)){
+            this.moveRight(player);
+            break;
+        }
+    }
+  }
+
+  validateMove(player, direction) {
+      const currPosition = (player == 1? this.p1.coords : this.p2.coords) 
+      let nextSpot
+    switch (direction) {
+      case "up":
+          nextSpot = this.matrix[currPosition.y - 1]; 
```

Comment:
You are assigning value to `nexrSpot` and in the next line overriding it...
You should assign the start value of false, and change the value to true depends on the condition